### DISCUSSION TITLE
Spectator/fly while in vanish

### DISCUSSION THREAD

**Discussion by VexiusNetworkUS** [Zrips/CMI#4476] [Feature Requests]
We are now using cmi vanish, we don't give all our staff the option to fly or go into spectator mode. The previous vanish plugin allowed double shift which put you into spectator mode and they were able to fly, this disabled as soon as they came out of vanish. Is is possible for this to be added as an option.

**Comment by CloudeLecaw** [comment]
What about a custom alias that is doing this?

**Comment by VexiusNetworkUS** [comment]
Maybe, it needs to turn on when they go into vanish and off when they come out of vanish.

**Comment by CloudeLecaw** [comment]
Lets say you have this in your langfile:
`  vanishSymbolOn: '&8[&7Vanish&8]&r'`

```
  vanish:
    Cmds:
    - perm:cmi.command.vanish#! check:%cmi_user_vanished_symbol%=!&8[&7Vanish&8]&r! cmi vanish [playername] on
    - check:%cmi_user_vanished_symbol%=!&8[&7Vanish&8]&r! cmi fly [playername] true -s
    - check:%cmi_user_vanished_symbol%==&8[&7Vanish&8]&r! cmi vanish [playername] off
    - check:%cmi_user_vanished_symbol%==&8[&7Vanish&8]&r! cmi fly [playername] false -s
```

what happens here:
when a player get into vanish he gets this symbol that is defined in he langfile, even if you dont use is on your tab or where somewhere else.
Now the command check if user symbol is equal to to this or not. if no (vanish is currently disabled) it executes vanish and fly true
if not (vanish is currently enabled)  it executed fly false and disables vanish
